server/src/main/java/org/elasticsearch/cluster/routing/allocation/Explanations.java

```diff
@@ -0,0 +1,108 @@
+            receives this information. Please wait.""";
+
+        public static final String ALL_COPIES_INVALID = """
+            Elasticsearch can't allocate this shard because all the copies of its data in the cluster are stale or corrupt. Elasticsearch \
+            will allocate this shard when a node containing a good copy of its data joins the cluster. If no such node is available, \
+            restore this index from a recent snapshot.""";
+
+        public static final String NO_COPIES = """
+            Elasticsearch can't allocate this shard because there are no copies of its data in the cluster. Elasticsearch will allocate \
+            this shard when a node holding a good copy of its data joins the cluster. If no such node is available, restore this index \
+            from a recent snapshot.""";
+
+        public static final String DELAYED_WITH_ALTERNATIVE = """
+            The node containing this shard copy recently left the cluster. Elasticsearch is waiting for it to return. If the node does not \
+            return within [%s] then Elasticsearch will allocate this shard to another node. Please wait.""";
+
+        public static final String DELAYED_WITHOUT_ALTERNATIVE = """
+            The node holding this shard copy recently left the cluster. Elasticsearch is waiting for it to return. If the node does not \
+            return within [%s] then Elasticsearch will attempt to allocate this shard to another node, but it cannot be allocated to any \
+            other node currently in the cluster. If you expect this shard to be allocated to another node, find this node in the \
+            node-by-node explanation and address the reasons which prevent Elasticsearch from allocating this shard there.""";
+
+        public static final String EXISTING_STORES_FORBIDDEN = """
+            Elasticsearch isn't allowed to allocate this shard to any of the nodes in the cluster that hold an in-sync copy of its data. \
+            Choose a node to which you expect this shard to be allocated, find this node in the node-by-node explanation, and address the \
+            reasons which prevent Elasticsearch from allocating this shard there.""";
+
+        public static final String ALL_NODES_FORBIDDEN = """
+            Elasticsearch isn't allowed to allocate this shard to any of the nodes in the cluster. Choose a node to which you expect this \
+            shard to be allocated, find this node in the node-by-node explanation, and address the reasons which prevent Elasticsearch \
+            from allocating this shard there.""";
+
+    }
+
+    public static final class Rebalance {
+
+        public static final String YES = """
+            Elasticsearch can rebalance this shard to another node.""";
+
+        public static final String ALREADY_BALANCED = """
+            Elasticsearch cannot rebalance this shard to another node since there is no node to which allocation is permitted which \
```

Comment:
I do think this is an improvement over the original messaging. But it seems to me this is the "normal good state" for shards and thus we could perhaps mention that here or reword this into a message that tones down the issue here? Something like "Elasticsearch believes this shard is in a well-balanced location according to allocation rules. If you expect this shard to be rebalanced to another node, find this node in the node-by-node explanation and address the reasons which prevent Elasticsearch from rebalancing this shard there."

Reply:
Thanks, that's a good point. See ea8c3cbed4749684ad81755c3a736dec63e2a109:

> This shard is in a well-balanced location and satisfies all allocation rules so
it will remain on this node. Elasticsearch cannot improve the cluster balance
by moving it to another node. If you expect this shard to be rebalanced to
another node, find the other node in the node-by-node explanation and address
the reasons which prevent Elasticsearch from rebalancing this shard there.